internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -131,6 +143,36 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 	return ctrl.Result{}, nil
 }
 
+// unstructuredNode returns a raw unstructured from a Node.
+func unstructuredNode(node *corev1.Node) *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(node.GroupVersionKind().GroupVersion().String())
+	obj.SetKind(node.Kind)
+	obj.SetName(node.Name)
+	obj.SetUID(node.UID)
+	obj.SetLabels(node.Labels)
+	return obj
+}
+
+// getManagedLabels gets a map[string]string and returns another map[string]string
+// filtering out labels not managed by CAPI.
+func getManagedLabels(labels map[string]string) map[string]string {
+	managedLabels := make(map[string]string)
+	for key, value := range labels {
+		if strings.HasSuffix(key, clusterv1.NodeRestrictionLabelDomain) {
+			managedLabels[key] = value
+		}
+		if strings.HasSuffix(key, clusterv1.ManagedNodeLabelDomain) {
```

Comment:
From the current state of the [proposal](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20220927-label-sync-between-machine-and-nodes.md) it looks like we want to sync all the labels that are in the `node.cluster.x-k8s.io` domain. Meaning a label like `node.cluster.x-k8s.io/foo = bar` should also be synced to the node. 
In this case a `HasSuffix` match will not catch these labels. 

Same for `node-restriction.kubernetes.io` domain labels.